include/dd/Export.hpp

```diff
@@ -39,10 +40,178 @@ namespace dd {
+                if (fraction.second == 1U) {
+                    os << fraction.first << "/√2";
+                } else {
+                    os << fraction.first << "/(" << fraction.second << "√2)";
+                }
+            }
+            return;
+        }
+
+        auto abspi = absr / PI;
+
+        if (std::abs(abspi - 1.) < tol) { // +- π
+            if (phase) {
+                os << " π)";
+            } else {
+                os << (std::signbit(r) ? "-" : "") << "π";
+            }
+            return;
+        }
+
+        fraction = ComplexValue::getLowestFraction(abspi);
+        approx   = static_cast<fp>(fraction.first) / static_cast<fp>(fraction.second);
+        error    = std::abs(abspi - approx);
+
+        if (error < tol) { // suitable fraction a/b π found
+            if (phase) {
+                if (fraction.second == 1U) {
+                    os << " " << fraction.first << "π)";
+                } else if (fraction.first == 1U) {
+                    os << " π/" << fraction.second << ")";
+                } else {
+                    os << " " << fraction.first << "π/" << fraction.second << ")";
+                }
+            } else {
+                if (fraction.second == 1U) {
+                    os << fraction.first << "π";
+                } else if (fraction.first == 1U) {
+                    os << "π/" << fraction.second;
+                } else {
+                    os << fraction.first << "π/" << fraction.second;
+                }
+            }
+            return;
+        }
+
+        if (phase) { // default
+            os << " " << absr << ")";
+        } else
+            os << r;
+    }
+
+    inline std::string asPolar(const Complex& a, bool formatted = true, int precision = -1) {
+        const auto         mag   = ComplexNumbers::mag(a);
+        const auto         phase = ComplexNumbers::arg(a);
+        std::ostringstream ss{};
+
+        if (precision >= 0) ss << std::setprecision(precision);
+
+        if (std::abs(mag) < ComplexTable<>::tolerance()) {
+            return "0";
+        }
+
+        if (formatted) {
+            if (std::abs(std::abs(phase) - dd::PI) < ComplexTable<>::tolerance()) {
+                ss << "-";
+                printPolarFormatted(ss, mag);
+            } else {
+                if (std::abs(mag - 1.0) > ComplexTable<>::tolerance()) {
+                    printPolarFormatted(ss, mag);
+                    if (std::abs(phase) > ComplexTable<>::tolerance()) {
+                        ss << " ";
+                    }
+                } else if (std::abs(phase) < ComplexTable<>::tolerance()) {
+                    ss << "1";
+                    return ss.str();
+                }
+                printPolarFormatted(ss, phase, true);
+            }
+        } else {
+            ss << mag;
+            if (std::abs(phase) > ComplexTable<>::tolerance()) {
+                ss << " " << phase;
+            }
+        }
+        return ss.str();
+    }
+
```

Comment:
Would it make sense to allow the User to switch between polar and rectangular coordinates? We could just use a Boolean input parameter to toggle either representation. I would guess that Polar Coordinates should be the default. This would also make it easy to incorporate such a switch into DDVis.